### PR TITLE
Adding in dev tooling to the contribution process

### PR DESCRIPTION
**Pull request by MWhite-22**
PR for issue #39

Happy to walk through individual commits with more detail

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,33 +2,30 @@
   "name": "create-t3-app",
   "version": "4.0.5",
   "description": "Create web application with the t3 stack",
+  "author": "Shoubhit Dash <youremail@here.com>",
```
Comment (MWhite-22) [inline package.json:5]: @nexxeln Be sure to add your email in here as I believe GitHub actually uses the information

Comment (staaph) [reply package.json:5]: shoubhit2005@gmail.com

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,46 @@
...
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+
+# MISC
+.idea
+.turbo/
+.cache/
+.next/
+.nuxt/
```
Comment (staaph) [inline .gitignore:38]: remove nuxt here to keep it clean

Comment (MWhite-22) [reply .gitignore:38]: Its just a generic .gitignore that covers 90% of use cases. If it doesn't hurt anything I would rather keep it complete and easily extensible to other projects.

File: package.json (hunk 1 above)
Comment (MWhite-22) [reply package.json]: Done

**Comment by staaph** [comment]
@nexxeln 
move prettier to editorconfig ?

**Comment by MWhite-22** [comment]
> @nexxeln move prettier to editorconfig ? https://editorconfig.org/

VSCode needs a plugin to work and I dont think this is a valid replacement for prettier (which works with eslint as well to prevent overriding rules re: styling)

**Comment by asheeeshh** [comment]
1. prettier -- it has been decided earlier that we don't need prettier either in the repository itself or in the generated project.
2. the same goes for eslint ig. @nexxeln your thoughts?
3. i personally don't know if we really need husky, let's see what others think about it

**Comment by MWhite-22** [comment]
1) The problem with not having a repo level prettier is that, as people contribute to the project, we will inevitably get a bloated git history of double quotes turning into single quotes, and vice versa, as well as other minor stylistic changes. If this project is expected to live on and grow as new technologies are integrated, setting a standard now, even if that standard changes, is a good idea IMO.

**Comment by MWhite-22** [comment]
> 1. prettier -- it has been decided earlier that we don't need prettier either in the repository itself or in the generated project. see #25 and #30

Both of these issues are about having prettier in the template built by create-t3, not having prettier IN create-t3. This PR is specifically for this repo, not adding prettier or eslint into any created templates

**Comment by nexxeln** [comment]
I would've liked it if we had discussed all the tools before you made a PR 😄 . 

### Prettier Config
Please change single quotes to *false*

### Eslint
This looks good to me

### Changeset
I've never used or heard of this package, what exactly will it do?

### Husky
I'm kind of 50/50 on this one. I would love to know what other people think.

**Comment by MWhite-22** [comment]
AFK for a few hours. Will push an update in a bit.

**Comment by nexxeln** [comment]
> Pushed new prettier config with double quotes.
Thank you, please format the files also with the new config.

Changesets seems cool so why not, but the first time I'll use it I'll poke you with questions on Discord 😄.

**Comment by nexxeln** [comment]
> Already re-formatted!

Oh lol sorry I didn't look at the commit.

I'm still unsure about husky.

**Comment by MWhite-22** [comment]
Totally happy to remove it if we need to. The current implementation just runs a lint and format before commit. 

Without it, it's up to people to lint and format on their own, and although most IDEs have built in settings to do so on save, not everyone is going have the same setups and extensions. 

This way we keep a relatively sane integrity check in the most "gentle" way possible.

**Comment by juliusmarminge** [comment]
I think it adds value as the amount of contributors grow.

Looking at commits and seeing a bunch of prettier-changes is annoying and makes it harder to see what the commit is actually doing.

What tool/setup to use for this I have no opinion on.

**Comment by nexxeln** [comment]
Okay let's do it then! I'll merge

**Pull request merged**